pkg/webhook/validate.go

```diff
@@ -200,13 +203,29 @@ func dynamicValidateSriovNetworkNodePolicy(cr *sriovnetworkv1.SriovNetworkNodePo
 }
 
 func validatePolicyForNodeStateAndPolicy(nsList *sriovnetworkv1.SriovNetworkNodeStateList, npList *sriovnetworkv1.SriovNetworkNodePolicyList, node *corev1.Node, cr *sriovnetworkv1.SriovNetworkNodePolicy) error {
+	nodeInterfaceErrorList := make(map[string][]string)
+	interfaceSelected = false
 	for _, ns := range nsList.Items {
 		if ns.GetName() == node.GetName() {
-			if err := validatePolicyForNodeState(cr, &ns, node); err != nil {
+			interfaceAndErrorList, err := validatePolicyForNodeState(cr, &ns, node)
+			if err != nil {
 				return err
 			}
+			if interfaceAndErrorList != nil {
+				nodeInterfaceErrorList[ns.GetName()] = interfaceAndErrorList
+			}
+		}
+		break
+	}
+
+	if !interfaceSelected {
```

Comment:
Shouldn't this block of code be next to `if !nodesSelected {` in the `dynamicValidateSriovNetworkNodePolicy` function?
This function handles a single node. This is why in your logs, the printing of the map for all nodes is not contiguous. The past behaviour is if none of the nodes match. In your current implementation it will print out even if 1 node is matching at the end.

Reply:
I agree! I fixed it and we can resolve now.